pkg/deploy/service_account.go

```diff
@@ -18,7 +18,12 @@ import (
 
 func SyncServiceAccountToCluster(deployContext *DeployContext, name string) (bool, error) {
 	saSpec := getServiceAccountSpec(deployContext, name)
-	return CreateIfNotExists(deployContext, saSpec)
+	isCreated, err := CreateIfNotExists(deployContext, saSpec)
+	if err != nil {
+		return false, err
+	}
+	// we're done if SA already exist
+	return !isCreated, nil
```

Comment:
```suggestion
	return err == nil, nil
```
After that we must take into account sync result (where this func is invoked).

Reply:
Sorry, I did not get your point fully. Error here is always nil, since we done check above.